### PR TITLE
Improve softmax loss docstrings

### PR DESCRIPTION
**Pull request by oxinabox**
Minor documentation change.
Because I forget that these things took kwargs only now

### CODE REVIEW

File: src/ops/nn.jl [hunk 1]
```diff
@@ -220,6 +220,9 @@ output of `softmax`, as it will produce incorrect results.
 `logits` and `labels` must have the same shape `[batch_size, num_classes]`
 and the same dtype (either `float32` or `float64`).
 
+**Note that to avoid confusion, it is required to pass only named arguments to
```
Comment (malmaud) [inline src/ops/nn.jl]: Let's tweak the grammar a bit. How about something like:

This function only accepts keyword arguments to avoid the frequent confusion that positional arguments cause.
